### PR TITLE
Allow course student to be upgraded to staff within course

### PR DESCRIPTION
**Pull request by kxmbrian**
For #677

### CODE REVIEW

File: app/controllers/concerns/course/users_controller_management_concern.rb [hunk 1]
```diff
@@ -31,9 +31,20 @@ def students # :nodoc:
   end
 
   def staff # :nodoc:
+    @student_options =
+      @course_users.students.with_approved_state.order_alphabetically.pluck(:name, :id)
+    @staff_role_options = CourseUser::STAFF_ROLES
```
Comment (allenwq) [inline app/controllers/concerns/course/users_controller_management_concern.rb]: Move this to view ? cause in future this need to be translated too.

File: spec/controllers/course/users_controller_spec.rb [hunk 2]
```diff
@@ -165,5 +165,30 @@
...
+
+      context 'when a course user cannot be added as staff' do
+        before do
+          controller.instance_variable_set(:@course_user, staff_to_be_stub)
+          subject
+        end
+
+        it { is_expected.to redirect_to(course_users_staff_path(course)) }
+        it 'sets an error flash message' do
+          expect(flash[:danger]).to eq('')
```
Comment (allenwq) [inline spec/controllers/course/users_controller_spec.rb]: why the message is empty ?

**Comment by allenwq** [comment]
👍

**Comment by weiqingtoh** [comment]
I kind of feel that `upgrade_to_staff` seems a little too specific as a controller action, maybe `update_role` might work. What do you all think?

**Comment by allenwq** [comment]
I have the discussion with @kxmbrian yesterday, think he made the action name specific because he want to differentiate with the `update` action. Maybe we should discuss again to clear this ?

**Comment by lowjoel** [comment]
Typically, how this is done in Rails is by having the update action deal with this. The permission can be a model property or something that triggers the transition.

Might not necessarily fix now, but Rails controllers vehemently stick to REST principles.

**Pull request merged**